doc/stp/SONiC_PVST_HLD.md

```diff
@@ -0,0 +1,700 @@
+
+
+
+### Local STP Port id allocation:
+
+For allocating local port ids STPd needs to know the max number of physical ports and Port-channel interfaces supported on the device. Currently there is no mechanism to obtain these values in a direct way (like reading from DBs) so below logic will be used to arrive at max ports -
+
+Max STP Ports = Max physical ports + Max Port-Channel interfaces 
+
+where,
+
+Max physical ports will be determined from the higher interface id populated say for example if higher port is Ethernet252 (or 254) - max ports will be 256 to ensure breakout ports are accomodated.
+
+Max Port-Channel interfaces will be same as max physical ports to accomodate the worst case scenario of each Port-channel having only one port.
+
+
+Post bootup, Portsyncd reads the port_config.ini file which contains the port specific configuration information, and updates the APP DB with this information. Orchagent listens to these APP DB updates and takes care of creating all the interfaces in linux kernel and the hardware. Kernel sends netlink updates for the interfaces being created which Portsyncd listens to, once Portsyncd confirms all the interfaces have been configured it will generate PortInitDone notification. STPd will receive this message via STPMgr and then gets all the interface information via netlink for allocating the local STP port ids.
+
+Note: The port id for Port-channel interface will be allocated only when the first member port is added, this ensures Port-channel interfaces with no member ports dont burn the port ids. 
+
+Example of port id allocation -
+```
+SONiC interface          STP Port id
+Ethernet0                    0
+Ethernet4                    4
+Ethernet8                    8
+Ethernet252                 252
+PortChannel10               256
+PortChannel5                257
+PortChannel20               258
+```
+
+Note: As Port-channel port ids are dynamically allocated it might result in same port channel getting different value post reboot. To ensure a predictable convergence where port-channel port id is used as tie breaker, user needs to configure appropriate port priority value.
+
+### 3.4.1 BPDU trap mechanism
+SONiC uses copp configuration file 00-copp.config.json for configuring the trap group, ids, cpu priority queue and a policer. This functionality has been extended for supporting STP and PVST BPDU trap to CPU.
+
+## 3.5 SAI
+STP SAI interface APIs are already defined and is available at below location -
+
+https://github.com/opencomputeproject/SAI/blob/master/inc/saistp.h
+
+## 3.6 CLI
```

Comment:
can we have a section to describe how to enable and disable this feature? looks like this is a part that most community concern is coming from.

Reply:
3.6.2.1.1 Global spanning-tree mode - This section describes how to enable or disable this feature, we can change the description to "Enabling/disabling of PVST feature" to make it clear.

Reply:
Document updated